openfisca_core/taxbenefitsystems.py

```diff
@@ -102,42 +98,115 @@ def new_scenario(self):
     def prefill_cache(self):
         pass
 
+    def load_variable(self, variable_class, update = False):
+        name = unicode(variable_class.__name__)
+        variable_type = variable_class.__bases__[0]
+        attributes = variable_class.__dict__
+
+        existing_column = self.get_column(name)
+
+        if existing_column and not update:
+            # Variables that are dependencies of others (trough a conversion column)can be loaded automatically
+            if name in self.automatically_loaded_variable:
+                self.automatically_loaded_variable.remove(name)
+                return self.get_column(name)
+            raise Exception("Variable {} is already defined. Use `update_variable` to replace it.".format(name))
+
+        if not existing_column and update:
+            raise Exception("Variable {} doesn't exist and can't be updated".format(name))
+
+        # We pass the variable_class just for introspection for parsers.
+        variable = variable_type(name, attributes, variable_class)
+        # We need the tax benefit system to identify columns mentioned by reference or PersonToEntityColumn...
+        column = variable.to_column(self)
+        self.column_by_name[column.name] = column
+
+        return column
+
+    def add_variable(self, variable_class):
+        return self.load_variable(variable_class, update = False)
+
+    def update_variable(self, variable_class):
+        return self.load_variable(variable_class, update = True)
```

Comment:
Note : We should check if there is a reference attribute given with the new variable. If so, we should replace the indicated variable and not be based on the name.